### DISCUSSION TITLE
Contributor License Agreement becoming mandatory for Saleor

### DISCUSSION THREAD

**Discussion by patrys** [saleor/saleor#7176] [Announcements]
Dear Contributors,

My name is Patryk Zawadzki and I am the CTO of Saleor Commerce.

We’ve spent a lot of time nurturing Saleor into the wonderful project it is today and making sure it enables its community to build great e-commerce experiences. We believe it is our responsibility to make sure you can continue to enjoy Saleor for years to come.

Today, we’re introducing an additional step to the contribution process. When making your next contribution, you will be asked to accept our Contributor License Agreement (the “CLA”) that gives Saleor the right to distribute your contribution and protects Saleor and its community from any liabilities caused by such a contribution.

Of course, this protection comes at the cost of having to deal with the CLA acceptance process. I’m an engineer myself, and I’m the first to admit that CLAs are a pain to deal with, but we strongly believe it’s the best way to ensure Saleor remains the reliable project you all expect it to be.

Thank you all for your amazing work. Saleor would be nothing without its community.  If you have any questions regarding the CLA, don't hesitate to let us know.

Best,

Patryk Zawadzki
CTO
Saleor Commerce

**Comment by benmehlman** [comment]
Does the CLA assign copyright?  If so, does it have a provision preventing contributed code from being released as part of a non-FOSS product?

**Reply by patrys** [reply]
It does assign copyright, and there is no such provision. We are open-source at heart and have no plans to change what's published under the permissive license. In this sense, anything you contribute will be part of an open-source release. At the same time, we are a company, and we want to control the IP we build and release, and we want to have the option to adjust future licensing to protect from exploitation and abuse. The vast majority of Saleor's code gets built in-house, and we're employing an entire team of engineers to maintain Saleor for everyone to enjoy, so I don't think that's too much to ask in return.
